### ISSUE TITLE
FilterError: /bin/sh: lessc: command not found

### ISSUE THREAD

**Issue by natea** [django-oscar/django-oscar-mws#4]
You might want to mention that "less" is a dependency:

I resolved the issue by running this:

```
$ npm install -g less
```

Here was the traceback:

```
Internal Server Error: /
Traceback (most recent call last):
  File "/Users/nateaune/.virtualenvs/django-oscar-mws/lib/python2.7/site-packages/django/core/handlers/base.py", line 137, in get_response
    response = response.render()
  File "/Users/nateaune/.virtualenvs/django-oscar-mws/lib/python2.7/site-packages/django/template/response.py", line 105, in render
    self.content = self.rendered_content
  File "/Users/nateaune/.virtualenvs/django-oscar-mws/lib/python2.7/site-packages/django/template/response.py", line 82, in rendered_content
    content = template.render(context)
  File "/Users/nateaune/.virtualenvs/django-oscar-mws/lib/python2.7/site-packages/django/template/base.py", line 140, in render
    return self._render(context)
  File "/Users/nateaune/.virtualenvs/django-oscar-mws/lib/python2.7/site-packages/django/template/base.py", line 134, in _render
    return self.nodelist.render(context)
  File "/Users/nateaune/.virtualenvs/django-oscar-mws/lib/python2.7/site-packages/django/template/base.py", line 840, in render
    bit = self.render_node(node, context)
  File "/Users/nateaune/.virtualenvs/django-oscar-mws/lib/python2.7/site-packages/django/template/debug.py", line 78, in render_node
    return node.render(context)
  File "/Users/nateaune/.virtualenvs/django-oscar-mws/lib/python2.7/site-packages/django/template/loader_tags.py", line 123, in render
    return compiled_parent._render(context)
  File "/Users/nateaune/.virtualenvs/django-oscar-mws/lib/python2.7/site-packages/django/template/base.py", line 134, in _render
    return self.nodelist.render(context)
  File "/Users/nateaune/.virtualenvs/django-oscar-mws/lib/python2.7/site-packages/django/template/base.py", line 840, in render
    bit = self.render_node(node, context)
  File "/Users/nateaune/.virtualenvs/django-oscar-mws/lib/python2.7/site-packages/django/template/debug.py", line 78, in render_node
    return node.render(context)
  File "/Users/nateaune/.virtualenvs/django-oscar-mws/lib/python2.7/site-packages/django/template/loader_tags.py", line 123, in render
    return compiled_parent._render(context)
  File "/Users/nateaune/.virtualenvs/django-oscar-mws/lib/python2.7/site-packages/django/template/base.py", line 134, in _render
    return self.nodelist.render(context)
  File "/Users/nateaune/.virtualenvs/django-oscar-mws/lib/python2.7/site-packages/django/template/base.py", line 840, in render
    bit = self.render_node(node, context)
  File "/Users/nateaune/.virtualenvs/django-oscar-mws/lib/python2.7/site-packages/django/template/debug.py", line 78, in render_node
    return node.render(context)
  File "/Users/nateaune/.virtualenvs/django-oscar-mws/lib/python2.7/site-packages/django/template/loader_tags.py", line 123, in render
    return compiled_parent._render(context)
  File "/Users/nateaune/.virtualenvs/django-oscar-mws/lib/python2.7/site-packages/django/template/base.py", line 134, in _render
    return self.nodelist.render(context)
  File "/Users/nateaune/.virtualenvs/django-oscar-mws/lib/python2.7/site-packages/django/template/base.py", line 840, in render
    bit = self.render_node(node, context)
  File "/Users/nateaune/.virtualenvs/django-oscar-mws/lib/python2.7/site-packages/django/template/debug.py", line 78, in render_node
    return node.render(context)
  File "/Users/nateaune/.virtualenvs/django-oscar-mws/lib/python2.7/site-packages/django/template/loader_tags.py", line 62, in render
    result = block.nodelist.render(context)
  File "/Users/nateaune/.virtualenvs/django-oscar-mws/lib/python2.7/site-packages/django/template/base.py", line 840, in render
    bit = self.render_node(node, context)
  File "/Users/nateaune/.virtualenvs/django-oscar-mws/lib/python2.7/site-packages/django/template/debug.py", line 78, in render_node
    return node.render(context)
  File "/Users/nateaune/.virtualenvs/django-oscar-mws/lib/python2.7/site-packages/django/template/loader_tags.py", line 62, in render
    result = block.nodelist.render(context)
  File "/Users/nateaune/.virtualenvs/django-oscar-mws/lib/python2.7/site-packages/django/template/base.py", line 840, in render
    bit = self.render_node(node, context)
  File "/Users/nateaune/.virtualenvs/django-oscar-mws/lib/python2.7/site-packages/django/template/debug.py", line 78, in render_node
    return node.render(context)
  File "/Users/nateaune/.virtualenvs/django-oscar-mws/lib/python2.7/site-packages/django_compressor-1.3-py2.7.egg/compressor/templatetags/compress.py", line 147, in render
    return self.render_compressed(context, self.kind, self.mode, forced=forced)
  File "/Users/nateaune/.virtualenvs/django-oscar-mws/lib/python2.7/site-packages/django_compressor-1.3-py2.7.egg/compressor/templatetags/compress.py", line 107, in render_compressed
    rendered_output = self.render_output(compressor, mode, forced=forced)
  File "/Users/nateaune/.virtualenvs/django-oscar-mws/lib/python2.7/site-packages/django_compressor-1.3-py2.7.egg/compressor/templatetags/compress.py", line 119, in render_output
    return compressor.output(mode, forced=forced)
  File "/Users/nateaune/.virtualenvs/django-oscar-mws/lib/python2.7/site-packages/django_compressor-1.3-py2.7.egg/compressor/css.py", line 51, in output
    ret.append(subnode.output(*args, **kwargs))
  File "/Users/nateaune/.virtualenvs/django-oscar-mws/lib/python2.7/site-packages/django_compressor-1.3-py2.7.egg/compressor/css.py", line 53, in output
    return super(CssCompressor, self).output(*args, **kwargs)
  File "/Users/nateaune/.virtualenvs/django-oscar-mws/lib/python2.7/site-packages/django_compressor-1.3-py2.7.egg/compressor/base.py", line 246, in output
    content = self.filter_input(forced)
  File "/Users/nateaune/.virtualenvs/django-oscar-mws/lib/python2.7/site-packages/django_compressor-1.3-py2.7.egg/compressor/base.py", line 194, in filter_input
    for hunk in self.hunks(forced):
  File "/Users/nateaune/.virtualenvs/django-oscar-mws/lib/python2.7/site-packages/django_compressor-1.3-py2.7.egg/compressor/base.py", line 169, in hunks
    precompiled, value = self.precompile(value, **options)
  File "/Users/nateaune/.virtualenvs/django-oscar-mws/lib/python2.7/site-packages/django_compressor-1.3-py2.7.egg/compressor/base.py", line 217, in precompile
    **kwargs)
  File "/Users/nateaune/.virtualenvs/django-oscar-mws/lib/python2.7/site-packages/django_compressor-1.3-py2.7.egg/compressor/filters/base.py", line 131, in input
    raise FilterError(err)
FilterError: /bin/sh: lessc: command not found
```

**Comment by roadsideseb** [comment]
Your issue is caused by the use of django-compressor in your project, as far as I can tell from the stacktrace. Since you are using Oscar and using compressor, it will try to compile the static file using the LESS compiler. The setup process for Oscar with LESS is describe [in the docs there](http://django-oscar.readthedocs.org/en/latest/internals/contributing/development-environment.html?highlight=less#writing-less-css). However, you don't have to use LESS for the static files unless you have a need for it in you own project, otherwise you can simply use the generated CSS that comes with Oscar.

I'm closing this issue because the project doesn't include any CSS files and doesn't depend on the use of compressor. These requirements are only by way of depending on Oscar.

**Comment by natea** [comment]
The reason that I mention this is because the first-time user experience for someone trying out django-oscar-mws, they are likely to run into this problem, since your sandbox app is using django-compressor: https://github.com/tangentlabs/django-oscar-mws/blob/master/sandbox/sandbox/settings.py#L150

**Comment by roadsideseb** [comment]
You're absolutely right. I didn't think of that. I'll write up some basic docs for it.

**Comment by natea** [comment]
@elbaschid would you recommend satisfying this dependency using your virtual-less package? https://github.com/elbaschid/virtual-less

**Comment by roadsideseb** [comment]
There are two ways of dealing with it inside of your virtualenv (if you don't want to install it globally). Either _virtual-less_ or [nodeenv](https://github.com/ekalinin/nodeenv). The main difference is that you'll have to run `nodeenv -p` manually in your virtualenv to install node, virtual-less will install it automatically when running `pip install virtual-less`. In general, it comes down to personal preference and I've used both succesfully.

I've also just looked at the sandbox of MWS and clean it up. The `lessc` exception above has been caused by `USE_LESS=True` in the setting which will use LESS but I don't think that's the best approach. I've switched it around in 84614cd to use the pre-compiled CSS instead. I've also added a section to the docs.
